pkg/ebpf/manager.go

```diff
@@ -74,11 +74,13 @@ type Modifier interface {
 
 // InitWithOptions is a wrapper around ebpf-manager.Manager.InitWithOptions
 func (m *Manager) InitWithOptions(bytecode io.ReaderAt, opts *manager.Options) error {
-	// we must load the ELF file before initialization,
-	// to build the collection specs, because some modifiers
-	// inspect these to make changes to the eBPF resources.
-	if err := m.LoadELF(bytecode); err != nil {
-		return fmt.Errorf("failed to load elf from reader: %w", err)
+	if bytecode != nil {
+		// we must load the ELF file before initialization,
+		// to build the collection specs, because some modifiers
+		// inspect these to make changes to the eBPF resources.
+		if err := m.LoadELF(bytecode); err != nil {
+			return fmt.Errorf("failed to load elf from reader: %w", err)
+		}
 	}
```

Comment:
In what flow we have nil `bytecode`?

Reply:
How is this change related to the current PR?

Reply:
This is because we are explicitly calling `LoadELF` in NPM now, so that the perf buffer upgrades can happen. Those upgrades require access to the map and program specs.

Reply:
> replicate the same elf state management from the ebpf-manager here

I'm not sure I follow what you are proposing?

Reply:
I was proposing to introduce a `LoadELF` function to the agent manager, and tracking whether the elf file has been loaded or not as part of the manager state. That way a user cannot mess up and call `LoadELF` and then pass a non-nil bytecode to the `InitWithOptions` function.

Reply:
I'm using the `Modifier` API now